lib/svg_sprite/svg_sprite.rb

```diff
@@ -0,0 +1,218 @@
+    "far-circle",
+    "far-clipboard",
+    "far-clock",
+    "far-comment",
+    "far-dot-circle",
+    "far-edit",
+    "far-envelope",
+    "far-eye",
+    "far-eye-slash",
+    "far-file-alt",
+    "far-frown",
+    "far-heart",
+    "far-image",
+    "far-list-alt",
+    "far-smile",
+    "far-square",
+    "far-thumbs-down",
+    "far-thumbs-up",
+    "far-trash-alt",
+    "fast-backward",
+    "fast-forward",
+    "file",
+    "file-alt",
+    "flag",
+    "folder",
+    "forward",
+    "gavel",
+    "globe",
+    "globe-americas",
+    "heading",
+    "heart",
+    "info-circle",
+    "italic",
+    "key",
+    "link",
+    "list",
+    "list-ol",
+    "list-ul",
+    "lock",
+    "microphone-slash",
+    "minus",
+    "mobile-alt",
+    "paint-brush",
+    "pencil-alt",
+    "plug",
+    "plus",
+    "power-off",
+    "question",
+    "question-circle",
+    "quote-left",
+    "quote-right",
+    "reply",
+    "search",
+    "share",
+    "shield-alt",
+    "signal",
+    "sign-out-alt",
+    "step-backward",
+    "step-forward",
+    "sync",
+    "table",
+    "tasks",
+    "thumbtack",
+    "times",
+    "times-circle",
+    "trash-alt",
+    "undo",
+    "unlock",
+    "upload",
+    "user",
+    "user-plus",
+    "users",
+    "wrench"
+  ])
+
+  FA_ICON_MAP = { 'far fa-' => 'far-', 'fab fa-' => 'fab-', 'fa-' => '' }
+
+  def self.all_icons
+    SVG_ICONS.merge(SiteSetting.svg_icon_subset.split('|'))
+    SVG_ICONS.merge(DiscoursePluginRegistry.svg_icons)
+    SVG_ICONS.merge(badge_icons)
+    SVG_ICONS.merge(group_icons)
+    SVG_ICONS.merge(theme_icons)
+  end
+
+  def self.bundle
+    require 'nokogiri'
+
+    icons = all_icons
+
+    @svg_subset = """
+      <!--
+      Font Awesome Free 5.4.1 by @fontawesome - https://fontawesome.com
+      License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
+      -->
+      <svg xmlns='http://www.w3.org/2000/svg' style='display: none;'>
+    """
+
+    Dir["#{Rails.root}/vendor/assets/svg-icons/fontawesome/*.svg"].each do |fname|
+      svg_file = Nokogiri::XML(File.open(fname)) do |config|
+        config.options = Nokogiri::XML::ParseOptions::NOBLANKS
+      end
+
+      svg_filename = "#{File.basename(fname, ".svg")}"
+
+      svg_file.css('symbol').each do |sym|
+        icon_id = sym.attr('id')
+
+        case svg_filename
+        when "regular"
+          icon_id = icon_id.prepend('far-')
+        when "brands"
+          icon_id = icon_id.prepend('fab-')
+        end
+
+        if icons.include? icon_id
+          sym.attributes['id'].value = icon_id
+          sym.css('title').each { |t| t.remove }
+          @svg_subset << sym.to_xml
+        end
+      end
+    end
+
+    @svg_subset << '</svg>'
+  end
+
+  def self.version(svg_subset)
+    (@svg_subset_cache ||= {})[svg_subset] ||=
+      Digest::SHA1.hexdigest(svg_subset)
+  end
+
+  def self.path
+    "/svg-sprite/#{Discourse.current_hostname}/#{version all_icons.to_s}.svg"
```

Comment:
if it's not a typo and version is a method I would add parentheses here: version(all_icons.to_s)